web/projects/ui/src/app/services/state.service.ts

```diff
@@ -0,0 +1,116 @@
+import { inject, Injectable } from '@angular/core'
+import { CanActivateFn, IsActiveMatchOptions, Router } from '@angular/router'
+import { ALWAYS_TRUE_HANDLER } from '@taiga-ui/cdk'
+import { TuiAlertService, TuiNotification } from '@taiga-ui/core'
+import {
+  BehaviorSubject,
+  combineLatest,
+  concat,
+  exhaustMap,
+  from,
+  merge,
+  Observable,
+  retry,
+  startWith,
+} from 'rxjs'
+import {
+  filter,
+  map,
+  shareReplay,
+  skip,
+  switchMap,
+  takeUntil,
+  tap,
+} from 'rxjs/operators'
+import { RR } from 'src/app/services/api/api.types'
+import { ApiService } from 'src/app/services/api/embassy-api.service'
+import { ConnectionService } from 'src/app/services/connection.service'
+
+const OPTIONS: IsActiveMatchOptions = {
+  paths: 'subset',
+  queryParams: 'exact',
+  fragment: 'ignored',
+  matrixParams: 'ignored',
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StateService extends Observable<RR.ServerState | null> {
```

Comment:
Why `| null`?

Reply:
`null` means we do not know the state yet. If it was just initial query — we would just wait for the response, but we cache the last response with `shareReplay` and if we lose connection to the server we need to be able to reset state to something that means "we currently don't know".